Sources/RSWeb/DownloadSession.swift

```diff
@@ -104,10 +104,21 @@ extension DownloadSession: URLSessionTaskDelegate {
 		
 		if response.statusCode == 301 || response.statusCode == 308 {
 			if let oldURLString = task.originalRequest?.url?.absoluteString, let newURLString = request.url?.absoluteString {
-				cacheRedirect(oldURLString, newURLString)
+                guard oldURLString != newURLString  else {
```

Comment:
Would you rewrite this as `if oldURLString == newURLString {`? This would make it easier to understand for future people reading the code.